woocommerce/woocommerce-android
# Issue/4030 init payment meta data

Parent issue #4030

This PR 
- adds basic metadata to payment intent
-  wraps all params related to payment with PaymentInfo object as the number of params was getting out of hand
- moves "convert to the smallest currency unit" from PaymentManager to CreatePaymentAction - the main reason for this change was that PaymentInfo object is immutable so updating/setting the amount was not feasible. In the end I think it kind of make sense to keep it in  CreatePaymentAction  anyway.

I recommend following the commits - the changes are quite straighforward and mostly done by the IDE

To test:
 - Make sure your store is eligible for in person payments
1. Connect to a reader (watch out for [this known issue](https://github.com/woocommerce/woocommerce-android/issues/4465)) 
2. Open detail of an unpaid order in USD
3. Tap on Collect Payment button
4. Finish the payment
5. Open Stripe's dashboard and search for the payment
6. Notice the payment contains "metadata" section, example:
<img width="500" alt="Screenshot 2021-07-21 at 12 28 44" src="https://user-images.githubusercontent.com/2261188/126476480-71234100-31e3-4826-b129-eb81f6aad439.png">

Update release notes:
